.hooks/pre-push

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+cargo check --all-features
+if [ $? -ne 0 ]; then
+    echo "Pre-push hook failed: cargo check detected errors."
+    exit 1
+fi
+
+cargo clippy --workspace --all-features --all-targets -- -D warnings
+if [ $? -ne 0 ]; then
+    echo "Pre-push hook failed: cargo clippy detected warnings or errors."
+    exit 1
+fi
+
+cargo fmt --all --check
```

Comment:
I think we should put `cargo +nightly fmt --all --check` no? because of fmt is based on nightly no?

Reply:
Fixed. I had always used it without nightly, but it makes sense to me to add it. Thanks